Prometheus component fails to compile with esp-idf

### The problem

Compiling the prometheus component fails when using esp-idf:
```
Compiling .pioenvs/test/cmock/CMock/src/cmock.o
src/main.cpp:18:1: error: 'prometheus' does not name a type
 prometheus::PrometheusHandler *prometheus_prometheushandler;
 ^~~~~~~~~~
src/main.cpp: In function 'void setup()':
src/main.cpp:138:3: error: 'prometheus_prometheushandler' was not declared in this scope
   prometheus_prometheushandler = new prometheus::PrometheusHandler(web_server_base_webserverbase);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/main.cpp:138:38: error: 'prometheus' does not name a type
   prometheus_prometheushandler = new prometheus::PrometheusHandler(web_server_base_webserverbase);
                                      ^~~~~~~~~~
```

### Which version of ESPHome has the issue?

2023.9.3

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

2023.10.1

### What platform are you using?

ESP32-IDF

### Board

esp32dev

### Component causing the issue

prometheus

### Example YAML snippet

```yaml
esphome:
  name: test

esp32:
  board: esp32dev
  framework:
    type: esp-idf

logger:

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

network:
  enable_ipv6: true

web_server:
  port: 80
  local: true

prometheus:

sensor:
  - platform: uptime
    name: Uptime Sensor
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/esphome/components/prometheus/prometheus_handler.cpp
+++ b/esphome/components/prometheus/prometheus_handler.cpp
@@ -87,7 +87,7 @@ void PrometheusHandler::sensor_row_(AsyncResponseStream *stream, sensor::Sensor 
     stream->print(obj->get_unit_of_measurement().c_str());
     stream->print(F("\"} "));
     stream->print(value_accuracy_to_string(obj->state, obj->get_accuracy_decimals()).c_str());
-    stream->print('\n');
+    stream->print(F("\n"));
   } else {
     // Invalid state
     stream->print(F("esphome_sensor_failed{id=\""));
@@ -122,7 +122,7 @@ void PrometheusHandler::binary_sensor_row_(AsyncResponseStream *stream, binary_s
     stream->print(relabel_name_(obj).c_str());
     stream->print(F("\"} "));
     stream->print(obj->state);
-    stream->print('\n');
+    stream->print(F("\n"));
   } else {
     // Invalid state
     stream->print(F("esphome_binary_sensor_failed{id=\""));
@@ -156,7 +156,7 @@ void PrometheusHandler::fan_row_(AsyncResponseStream *stream, fan::Fan *obj) {
   stream->print(relabel_name_(obj).c_str());
   stream->print(F("\"} "));
   stream->print(obj->state);
-  stream->print('\n');
+  stream->print(F("\n"));
   // Speed if available
   if (obj->get_traits().supports_speed()) {
     stream->print(F("esphome_fan_speed{id=\""));
@@ -165,7 +165,7 @@ void PrometheusHandler::fan_row_(AsyncResponseStream *stream, fan::Fan *obj) {
     stream->print(relabel_name_(obj).c_str());
     stream->print(F("\"} "));
     stream->print(obj->speed);
-    stream->print('\n');
+    stream->print(F("\n"));
   }
   // Oscillation if available
   if (obj->get_traits().supports_oscillation()) {
@@ -175,7 +175,7 @@ void PrometheusHandler::fan_row_(AsyncResponseStream *stream, fan::Fan *obj) {
     stream->print(relabel_name_(obj).c_str());
     stream->print(F("\"} "));
     stream->print(obj->oscillating);
-    stream->print('\n');
+    stream->print(F("\n"));
   }
 }
 #endif
@@ -279,7 +279,7 @@ void PrometheusHandler::cover_row_(AsyncResponseStream *stream, cover::Cover *ob
     stream->print(relabel_name_(obj).c_str());
     stream->print(F("\"} "));
     stream->print(obj->position);
-    stream->print('\n');
+    stream->print(F("\n"));
     if (obj->get_traits().get_supports_tilt()) {
       stream->print(F("esphome_cover_tilt{id=\""));
       stream->print(relabel_id_(obj).c_str());
@@ -287,7 +287,7 @@ void PrometheusHandler::cover_row_(AsyncResponseStream *stream, cover::Cover *ob
       stream->print(relabel_name_(obj).c_str());
       stream->print(F("\"} "));
       stream->print(obj->tilt);
-      stream->print('\n');
+      stream->print(F("\n"));
     }
   } else {
     // Invalid state
@@ -320,7 +320,7 @@ void PrometheusHandler::switch_row_(AsyncResponseStream *stream, switch_::Switch
   stream->print(relabel_name_(obj).c_str());
   stream->print(F("\"} "));
   stream->print(obj->state);
-  stream->print('\n');
+  stream->print(F("\n"));
 }
 #endif
 
@@ -344,7 +344,7 @@ void PrometheusHandler::lock_row_(AsyncResponseStream *stream, lock::Lock *obj) 
   stream->print(relabel_name_(obj).c_str());
   stream->print(F("\"} "));
   stream->print(obj->state);
-  stream->print('\n');
+  stream->print(F("\n"));
 }
 #endif
 

--- a/esphome/components/prometheus/prometheus_handler.cpp
+++ b/esphome/components/prometheus/prometheus_handler.cpp
@@ -1,5 +1,3 @@
-#ifdef USE_ARDUINO
-
 #include "prometheus_handler.h"
 #include "esphome/core/application.h"
 
@@ -352,5 +350,3 @@ void PrometheusHandler::lock_row_(AsyncResponseStream *stream, lock::Lock *obj) 
 
 }  // namespace prometheus
 }  // namespace esphome
-
-#endif  // USE_ARDUINO

--- a/esphome/components/prometheus/prometheus_handler.h
+++ b/esphome/components/prometheus/prometheus_handler.h
@@ -1,14 +1,12 @@
 #pragma once
 
-#ifdef USE_ARDUINO
-
 #include <map>
 #include <utility>
 
-#include "esphome/core/entity_base.h"
 #include "esphome/components/web_server_base/web_server_base.h"
-#include "esphome/core/controller.h"
 #include "esphome/core/component.h"
+#include "esphome/core/controller.h"
+#include "esphome/core/entity_base.h"
 
 namespace esphome {
 namespace prometheus {
@@ -119,5 +117,3 @@ class PrometheusHandler : public AsyncWebHandler, public Component {
 
 }  // namespace prometheus
 }  // namespace esphome
-
-#endif  // USE_ARDUINO

